include/sys/types.h

```diff
@@ -188,6 +188,9 @@ typedef intptr_t     ptrdiff_t;
 
 typedef uint16_t     wchar_t;
 #endif
+#if defined(__WCHAR_WIDTH__) && __WCHAR_WIDTH__ != 16
+#error __WCHAR_WIDTH__ should be 16
```

Comment:
who define __WCHAR_WIDTH__?

Reply:
it's a compiler predefined macro.

Reply:
But wchar_t is compiler defined value from wiki:
https://en.wikipedia.org/wiki/Wide_character

Reply:
yes.
the compiler knows wchar_t and thus can predefine `__WCHAR_WIDTH__`.
the problem here is that we are using compilers which don't know nuttx's wchar_t.
we can workaround it with -fshort-wchar.

Reply:
So, should you split the check to another patch, maybe you can print the related option too. But why gcc complain __WCHAR_WIDTH__ not equal to 16 for arm?

Reply:
do you mean to make this #warnings instead of #error?

i guess the compiler we are using for arm is configured for linux and thus `__WCHAR_WIDTH__` = 32.
anyway 32 is commonly used for modern OSes except windows.